dp/jobqueueing/gcsub.py

```diff
@@ -0,0 +1,164 @@
+"""
+Script to dequeue one or more generate_climos queue entries with NEW status, and submit a PBS job for each,
+updating the queue entries accordingly.
+
+Entries are dequeued in order of addition to the database; i.e., it is a FIFO queue.
+"""
+
+from argparse import ArgumentParser
+import datetime
+import logging
+import os.path
+import sys
+from subprocess import Popen, PIPE
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy import asc
+
+from dp.script_helpers import default_logger
+from dp.jobqueueing.argparse_helpers import add_global_arguments, add_submit_arguments
+from dp.jobqueueing.jobqueueing_db import GenerateClimosQueueEntry
+
+
+logger = default_logger()
+
+
+def make_qsub_test_script(queue_entry):
+    return '''
+#PBS -l nodes=1:ppn={qe.ppn}
+#PBS -l vmem={vmem}mb
+#PBS -l walltime={qe.walltime}
+#PBS -o {qe.output_directory}/logs
+#PBS -e {qe.output_directory}/logs
+#PBS -m abe
+#PBS -N generate_climos:{input_filename}
+#PBS -d /storage/home/rglover/code/climate-explorer-backend
+
+pbs_job_num=$(expr match "$PBS_JOBID" '\([0-9]*\)')
+
+# Set up the execution environment
+module load netcdf-bin
+module load cdo-bin
+source py3.4/bin/activate
```

Comment:
I do think that the standard environment (repo clone) should not be in my home directory. I'll move it to somewhere stable, e.g., `/storage/data/projects/comp_support/climate_exporer_data_prep/climatological_means/`